Fix: Last Played Media Title in plex would stay even when player was …

ryanm101 opened the pull request:
…idle/off

     Primary Fix is in the "if self._device" portion.
     code in "if self._session" is a catch all but i'm not 100% if it is needed.

## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

ryanm101 committed:
Fix: Last Played Media Title in plex would stay even when player was idle/off
     Primary Fix is in the "if self._device" portion.
     code in "if self._session" is a catch all but i'm not 100% if it is needed.

homeassistant commented:
Hi @ryanm101, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9664).

Once you do that we will be able to review and accept this pull request.

Thanks!

armills commented on homeassistant/components/media_player/plex.py:
```diff
@@ -347,6 +347,7 @@ def refresh(self, device, session):
         if device:
             self._device = device
             self._session = None
+            self._media_title = None
```
Some notes:
1) We should clear out all of the `self._media*` properties if we know nothing is playing.
2) Can we move this all to one location? I'm picturing something like this at the end of refresh:
```python
if not self._is_player_active:
    self._media_title = None
    self._media_image_url = None
    …
```

Also, this is a monster of a method and could use some refactoring, but that doesn't need to be part of this PR.

ryanm101 replied:
A quick test and commenting out my line under "device" and doing your not active resulted in the issue resurfacing.

Going redo my PR with linter fix and if you could approve that i'll look into refactoring the refresh in another PR

armills replied:
Strange. Well in any case we need to treat all of the media info properties the same. You might want to create a helper method to clear them all out and then call it in these locations.

ryanm101 replied:
Am I able to do that under another PR that way I can do a refactor on that refresh call and hopefully better understand it.

ryanm101 committed:
Fixed lint issues with previous commit

ryanm101 committed:
1st Pass at refactoring plex refresh
Moved _media** into clearMedia() which is called in _init_ and
at start of refresh.

Removed redunant _media_* = None entries

Grouped TV Show and Music under single if rather than testing
seperately for now.

ryanm101 committed:
Fixed invalid name for _clearMedia()
Removed another media_* = None entry

ryanm101 committed:
Removed print() statements used for debug

armills commented on homeassistant/components/media_player/plex.py:
```diff
@@ -418,65 +421,53 @@ def refresh(self, device, session):
-            self._media_album_artist = None
-            self._media_track = None
-            self._media_artist = None
+        if self._is_player_active:
```
Reading through the code, it looks to me like `self._media_content_type` will only be not None if `self._is_player_active` is True, so I think we can drop this outer if all-together. It would be good if that's the case to simplify this diff.

ryanm101 replied:
I'll do that today/tonight was away at a conf so no access to test. I had left it in because I wasnt sure on impact and and I want to clean up the working code 1st, i think a few line above that there was then entry for Movies so I did think that if was possibly to be removed else the Movies part was to come under that.. 

Also I've noticed in my case the IF SESSION: at the top is (seemingly) never triggered but if DEVICE is always triggered. So I'm not sure in which case it would get called.

emlove reviewed (commented):
Looks like a good first step to me! Just one small note.

ryanm101 committed:
Removed unneeded "if" statement

emlove reviewed (approved):
Nice! Looks like a solid start!

armills commented:
Thanks for the contribution! 🔥
